crates/ide/src/display/navigation_target.rs

```diff
@@ -502,21 +501,22 @@ impl TryToNav for hir::ConstParam {
 ///
 /// e.g. `struct Name`, `enum Name`, `fn Name`
 pub(crate) fn description_from_symbol(db: &RootDatabase, symbol: &FileSymbol) -> Option<String> {
-    let parse = db.parse(symbol.file_id);
-    let node = symbol.ptr.to_node(parse.tree().syntax());
+    let sema = Semantics::new(db);
```

Comment:
Would it make sense to pass `Semantics` here? Loos like it's available in some or most of the places where `to_nav` is called.

Reply:
Not really, the relevant place here doesn't have semantics either https://github.com/rust-analyzer/rust-analyzer/blob/979e788957ced1957ee9ac1da70fb97abf9fe2b1/crates/ide/src/display/navigation_target.rs#L167-L191

Reply:
I mean, `runnable_mod` has a `sema` on hand and calls `to_nav` on `Module`, but `symbol_search` doesn't. In other places like `find_all_refs` we pass `Semantics` to avoid recomputing it in some cases.

Reply:
Ah I see what you mean, I'll check it out

Reply:
I think it's fine for now, until other `ToNav` implementations start needing `Semantics`. On the other hand, I'm not sure what's the difference (`SemanticsImpl::parse` seems to do the same thing, but cache the result, which I don't think we need here?).

Reply:
It should be required here for `to_def` to work properly I think. I'm pretty sure it's gonna panic as the `SourceFile` node won't be indexed by semantics otherwise.